packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -9,8 +9,251 @@
+
+portfolio.giveAccess(user);
+portfolio.giveAccess(group);
+portfolio.giveAccess(role);
+```
+
+## Constraints
+
+Constraints define governance mechanisms that allow you to manage permissions and notifications relate to actions on your stacks. Constraints are added to a portfolio-product association. Using the cdk, if you do not explicity associate a product to a portfolio and add a constraint, it will automatically add the association for you. 
+
+There are rules around constraint configurations.  For example, you may not have more than one `Launch Role` or `Stack Set` constraint applied to a portfolio-product, and you may not have both of them on same assocation.  If a misconfigured constraint is added, `synth` will fail with an error message.giti 
+
+Read more at [Constraints](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/constraints.html)
+
+
+### Event Notifications
+
+An event update constraint allows user to specify a list of AWS `SNS` topics to receive notifications about stack events.
+
+```ts
+const portfolio = new Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+  });
+
+const product = new Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+const topics = [new Topic(stack, 'topic1'), new Topic(stack, 'topic2'), new Topic(stack, 'topic3')];
+
+portfolio.addEventNotifications({ snsTopics: topics, product: product });
+```
+
+### Launch Role
+
+A launch role constraint specifies a role that the end user assumes when launching a product. You cannot have more than one of these applied to a portfolio-product association
+
+```ts
+const portfolio = new Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+  });
+
+const product = new Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+const launchRole = new Role(this, 'LaunchRole', {
+  assumedBy: new AccountRootPrincipal(),
+});
+
+portfolio.addLaunchRole({
+  role: launchRole,
+  product: product,
+});
+
+```
+
+### Provisioning Rules
+
+Provisioning rules specify certain options available to end users when launching a product, such as an `instance type`.  Rules are specified as a JSON string.
+
+```ts
+const portfolio = new Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+  });
+
+const product = new Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+portfolio.addProvisioningRules({
```

Comment:
Yes that is going to come on a following PR, the issue there is that it will apply the rule to the scope of the current stack, whereas it should only apply to the template referenced in the link. For now, we are relying on this for the first PR so we can have some separation for iterative development.

Reply:
Can we remove the `addProvisioningRules()` method for now then? I'd rather not add a method that we know doesn't have a good API.